client/my-sites/checkout/composite-checkout/hooks/product-variants.tsx

```diff
@@ -62,26 +73,25 @@ const DoNotPayThis = styled.span`
 	}
 `;
 
-export function useProductVariants( {
-	siteId,
-	productSlug,
-}: {
-	siteId: number | undefined;
-	productSlug: string | undefined;
-} ): GetProductVariants {
+export function useGetProductVariants(
+	siteId: number | undefined,
+	productSlug: string
+): WPCOMProductVariant[] {
 	const translate = useTranslate();
 	const reduxDispatch = useDispatch();
 
+	const sitePlans: SitesPlansResult | null = useSelector( ( state ) =>
+		siteId ? getPlansBySiteId( state, siteId ) : null
+	);
+	const activePlan = sitePlans?.data?.find( ( plan ) => plan.currentPlan );
+	const isProductAnUpgrade = useSelector( ( state ) =>
+		siteId ? canUpgradeToPlan( state, siteId, productSlug ) : false
```

Comment:
Is the logic here actually correct?  I would expect to be checking each variant's product slug separately (rather than the overall `productSlug`).  I'm also not 100% sure that `canUpgradeToPlan()` is even necessary; regardless of whether it's an upgrade or not, do we **ever** want to allow a term downgrade via the variant picker specifically (even if we allow it in other contexts)?

I think one consequence of the above is the following behavior with this pull request:
1. Buy an annual plan.
2. Change the expiration date to something soon.
3. Go to the Manage Purchase page and choose the "Renew Monthly" option.
4. Open up the variant selector; notice that monthly is selected and all variants are available.
5. Switch to a 2-year plan; notice that monthly is no longer available, so you can't switch back.  (That's new behavior with this pull request.)
6. Switch to an annual plan; that turns the purchase into a renewal, so none of the variants are available to switch back to. (That's not ideal, but it is preexisting behavior.)

Also, if you repeat the above steps but start with a 2-year plan rather than an annual plan, the variant picker gives you a backdoor way to switch from 2-year to 1-year; that's existing behavior before this pull request too, although it's a downgrade path that we ideally shouldn't expose in the UI since our billing system isn't really ready for it.

If we just blocked all term downgrades via the variant picker (and then blocked the variant picker itself if the current item in the cart is a term downgrade) that might solve all the above, both the preexisting issues and the new one?

Reply:
> Switch to a 2-year plan; notice that monthly is no longer available, so you can't switch back. (That's new behavior with this pull request.)

Oh... that wasn't intentional. Downgrading terms for the _same_ plan was supposed to be allowed. In fact, that's why `canUpgradeToPlan` is there. I thought that was what we were supposed to be doing.

> If we just blocked all term downgrades via the variant picker (and then blocked the variant picker itself if the current item in the cart is a term downgrade) that might solve all the above, both the preexisting issues and the new one?

Sounds reasonable. I'll give that a shot!